### PR TITLE
Add Frequently Asked Questions 

### PR DESCRIPTION
**Pull request by awsalialem**
Added frequently asked questions:

- Timestamp validation errors and how to fix that
- Negative Offset lag and ways to fix that

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -298,3 +298,37 @@ MirrorMaker2 should size `tasks.max` so that each MirrorSourceConnector task has
 * **Why it matters:** The number of records per task shows the aggregate throughput that Kafka Connect is processing. Each MSC task should be balanced with a similar number of overall records, and any MSC task with many more or fewer records indicates an unbalanced MirrorMaker2 replication state. This likely indicates a hot partition, or an incorrect setting for `tasks.max`.
 
 
+### I am getting `InvalidRecordException` exception with timestamp out of range error
```
Comment (mmehrten) [inline README.md:301]: Kind of nitpicky - should this be "Why am I getting `InvalidRecordException: Timestamp ##### of message with offset ## is out of range` exception?"

File: README.md [hunk 2]
```diff
@@ -298,3 +298,37 @@ MirrorMaker2 should size `tasks.max` so that each MirrorSourceConnector task has
...
+
+* **What is the error:** This shows as an exception which causes the tasks fail and replication stops. It could happen right after you start the source connector, or when the connector has stopped for a long time and you restart it again. This error usually shows in logs as follows:
+
+```sh
```
Comment (mmehrten) [inline README.md:305]: Remove `sh` - this is plaintext so we don't need a language for syntax highlighting

File: README.md [hunk 3]
```diff
@@ -298,3 +298,37 @@ MirrorMaker2 should size `tasks.max` so that each MirrorSourceConnector task has
...
+Caused by: org.apache.kafka.common.InvalidRecordException: Timestamp 1716652875350 of message with offset 0 is out of range. The timestamp should be within [1716927328590, 1717100128590]
+
+```
+
+* **Root cause:** There is a broker level configuration `log.message.timestamp.difference.max.ms` that manages how the broker should validate the events' timestamp. If this configuration is set to a  value that is shorter than the age of the oldest message on a source topic, this error will occur.
+
+* **How to fix it:** If you expect to replicate the data which is older than the configured value, you would need to increase this value and restart your Kafka brokers. 
+
+
+### We are seeing negative offset lag when inspecting the `__consumer_offsets` topic for only a subset of partitions for a given consumer.
```
Comment (mmehrten) [inline README.md:326]: Same nitpick, should this be a question?

File: README.md [hunk 4]
```diff
@@ -298,3 +298,37 @@ MirrorMaker2 should size `tasks.max` so that each MirrorSourceConnector task has
...
+
+* **What is causing this error:** This bug occurs when using MirrorMaker2 with `sync.group.offsets.enabled = true`
+
+    If a source partition is empty, but the source consumer group's offset for that partition is non-zero, then MirrorMaker sets the target consumer group's offset for that partition to the literal, not translated, offset of the source consumer group. This state can be reached if the source consumer group consumed some records that were now deleted (like by a retention policy), or if MirrorMaker replication is set to start at "latest". This bug causes the target consumer group's lag for that partition to be negative and breaks offset sync for that partition until lag is positive.
+
+    The correct behavior when the source partition is empty would be to set the target offset to the translated offset, not literal offset, which in this case would always be 0. [More information](https://issues.apache.org/jira/browse/KAFKA-12635).
+
+* **What is the solution:** Manually resetting the consumer group or upgrading the Kafka Connect version to >=3.3 should fix this problem.
```
Comment (mmehrten) [inline README.md:334]: Manually resetting the source consumer group offset? Or something else?

Comment (mmehrten) [inline README.md:334]: Nitpick "should fix this problem" > "will fix this problem."

File: README.md [hunk 5]
```diff
@@ -298,3 +298,37 @@ MirrorMaker2 should size `tasks.max` so that each MirrorSourceConnector task has
...
+	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
+	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
+	at java.base/java.lang.Thread.run(Thread.java:829)
+Caused by: org.apache.kafka.common.InvalidRecordException: Timestamp 1716652875350 of message with offset 0 is out of range. The timestamp should be within [1716927328590, 1717100128590]
+
+```
+
+* **Root cause:** There is a broker level configuration `log.message.timestamp.difference.max.ms` that manages how the broker should validate the events' timestamp. If this configuration is set to a  value that is shorter than the age of the oldest message on a source topic, this error will occur.
+
+* **How to fix it:** If you expect to replicate the data which is older than the configured value, you would need to increase this value and restart your Kafka brokers.
```
Comment (mmehrten) [inline README.md:323]: How often is this true? Should we be updating the defaults or calling this out sooner?

**Review by mmehrten — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (awsalialem) [reply README.md:301]: updated

File: README.md (hunk 2 above)
Comment (awsalialem) [reply README.md:305]: updated

File: README.md (hunk 3 above)
Comment (awsalialem) [reply README.md:326]: updated

File: README.md (hunk 4 above)
Comment (awsalialem) [reply README.md:334]: Updated with more clear instructions

File: README.md (hunk 5 above)
Comment (awsalialem) [reply README.md:323]: This only happens if that configuration is set to non-default with a very short range. So should be called out only if this exception is received.

**Review by mmehrten — APPROVED** [review]

**Pull request merged**